torch/csrc/distributed/rpc/rref_impl.cpp

```diff
@@ -109,20 +109,28 @@ UserRRef<T>::UserRRef(
 }
 
 template <typename T>
-UserRRef<T>::~UserRRef() {
-  try {
-    RRefContext::getInstance().delUser(ownerId_, rrefId_, forkId_);
-  } catch (const std::exception& ex) {
-    LOG(ERROR) << "Error occurred when deleting UserRRef instance, "
-               << "RRefId = " << rrefId_ << ", ForkId = " << forkId_ << " : "
-               << ex.what();
-  } catch (...) {
-    LOG(ERROR) << "Error occurred when deleting UserRRef instance, "
-               << "RRefId = " << rrefId_ << ", ForkId = " << forkId_ << " : "
-               << "unknown error";
+void UserRRef<T>::tryDel() {
+  bool valid = valid_.exchange(false);
```

Comment:
This logic is not correct, we are setting valid to false before we actually send out the delFork message. We should do this after sending the delFork message. It could be possible that the delFork fails, but we have set valid to false, which is incorrect.

Reply:
@pritamdamania87 

Nice catch.

Notice that, that way, we might send out the delete user rref request twice.

One by the Python GC, the other by the shutdown call.

So I need to remove the assertion that limits this request to once too.